services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/SwapEndpointSpec.groovy

```diff
@@ -0,0 +1,887 @@
+
+        when: "Try to swap endpoints for two flows"
+        northbound.swapFlowEndpoint(
+                new SwapFlowPayload(flow1.id, flow1Src, flow1Dst),
+                new SwapFlowPayload(flow2.id, flow2Src, flow2Dst))
+
+        then: "An error is received (400 code)"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
+        exc.responseBodyAsString.to(MessageError).errorMessage == "Can not swap endpoints for flows: " +
+                "new requested endpoint for '$flow2.id' conflicts with existing endpoint for '$flow1.id'"
+
+        and: "Delete flows"
+        [flow1, flow2].each { flowHelper.deleteFlow(it.id) }
+
+        where:
+        description << ["the same src endpoint for flows", "the same dst endpoint for flows"]
+        flow1SwitchPair << [getTopologyHelper().getNotNeighboringSwitchPair()] * 2
+        flow2SwitchPair << [getDifferentNotNeighboringSwitchPair(flow1SwitchPair)] * 2
+        flow1 << [getFirstFlow(flow1SwitchPair, flow2SwitchPair)] * 2
+        flow2 << [getSecondFlow(flow1SwitchPair, flow2SwitchPair, flow1)] * 2
+        [flow1Src, flow1Dst, flow2Src, flow2Dst] << [
+                [flow1.source, changePropertyValue(flow1.destination, "portNumber", flow2.destination.portNumber),
+                 flow1.source, changePropertyValue(flow2.destination, "portNumber", flow1.destination.portNumber)],
+                [changePropertyValue(flow1.source, "portNumber", flow2.source.portNumber), flow1.destination,
+                 changePropertyValue(flow2.source, "portNumber", flow1.source.portNumber), flow1.destination]
+        ]
+    }
+
+    @Unroll
+    def "Unable to swap ports for two flows (port is occupied by ISL on #switchType switch)"() {
+        given: "Two active flows"
+        flowHelper.addFlow(flow1)
+        flowHelper.addFlow(flow2)
+
+        when: "Try to swap ports for two flows"
+        northbound.swapFlowEndpoint(
+                new SwapFlowPayload(flow1.id, flow1Src, flow1Dst),
+                new SwapFlowPayload(flow2.id, flow2Src, flow2Dst))
+
+        then: "An error is received (400 code)"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 400
+        exc.responseBodyAsString.to(MessageError).errorMessage == "Can not swap endpoints for flows: " +
+                "The port $islPort on the switch '${switchPair."$switchType".dpId}' is occupied by an ISL."
+
+        and: "Delete flows"
+        [flow1, flow2].each { flowHelper.deleteFlow(it.id) }
+
+        where:
+        switchType << ["src", "dst"]
+        switchPair << [getTopologyHelper().getNotNeighboringSwitchPair()] * 2
+        islPort << [getTopology().getBusyPortsForSwitch(switchPair.src).first(),
+                    getTopology().getBusyPortsForSwitch(switchPair.dst).first()]
+        flow1 << [getFlowHelper().randomFlow(switchPair)] * 2
+        flow2 << [
+                changePropertyValue(getFlowHelper().randomFlow(switchPair, false, [flow1]),
```

Comment:
I think it's more relevant to use `getFlowHelper().randomFlow(switchPair, false, [flow1]).tap {...}` here instead of 'changePropertyValue'

Reply:
why is that? Just do 
```
flow2 << [
  getFlowHelper().randomFlow(switchPair, false, [flow1]).tap { it.destination.portNumber = islPort }
   ....
]
```
There is no reason to create a 'copy' instance here